src/main/java/games/strategy/triplea/ai/proAI/util/ProPurchaseUtils.java

```diff
@@ -256,15 +257,10 @@ public static void incrementUnitProductionForBidTerritories(
   }
 
   private static int getUnitProduction(final Territory territory, final GameData data, final PlayerID player) {
-    final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.newCompositeBuilder(
-        Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player),
-        Matches.unitIsBeingTransported().invert());
-    if (territory.isWater()) {
-      factoryMatchBuilder.add(Matches.unitIsLand().invert());
-    } else {
-      factoryMatchBuilder.add(Matches.unitIsSea().invert());
-    }
-    final Collection<Unit> factoryUnits = territory.getUnits().getMatches(factoryMatchBuilder.all());
+    final Predicate<Unit> factoryMatch = Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player)
+        .and(Matches.unitIsBeingTransported().invert())
+        .and((territory.isWater() ? Matches.unitIsLand() : Matches.unitIsSea()).invert());
```

Comment:
I think you are missing an invert() on isLand here.

Reply:
Yeah, this tripped me up as well during the first review.  There's parentheses wrapping the entire ternary expression, so that `invert()` is applied to the value of that expression, not only to `Matches.unitIsSea()`.

Reply:
Ah. I see now. Missed the parenthesis.